scaleway/resource_instance_server_test.go

```diff
@@ -1404,3 +1404,43 @@ func TestAccScalewayInstanceServer_MigrateInvalidLocalVolumeSize(t *testing.T) {
 		},
 	})
 }
+
+func TestAccScalewayInstanceServer_CustomDiffImage(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayInstanceServerDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource "scaleway_instance_server" "main" {
+						image = "ubuntu_jammy"
+						type = "DEV1-S"
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayInstanceServerExists(tt, "scaleway_instance_server.main"),
+					resource.TestCheckResourceAttr("scaleway_instance_server.main", "image", "ubuntu_jammy"),
+				),
+			},
+			{
+				Config: `
+					data "scaleway_marketplace_image" "jammy" {
+						label = "ubuntu_jammy"
+					}
+					resource "scaleway_instance_server" "main" {
+						image = data.scaleway_marketplace_image.jammy.id
+						type = "DEV1-S"
+					}
+				`,
+				ExpectNonEmptyPlan: false,
```

Comment:
`ExpectNonEmptyPlan` is false per default. You should look into `PlanOnly` to ensure that the plan is empty.
Here the test could be successful if the instance is re-created.